locator/load_sketch.hh

```diff
@@ -137,6 +137,17 @@ public:
         std::make_heap(n._shards.begin(), n._shards.end(), shard_load_cmp());
     }
 
+    void pick(host_id node, shard_id shard) {
+        auto& n = _nodes.at(node);
+        for (auto& shard_load : n._shards) {
+            if (shard_load.id == shard) {
+                ++shard_load.load;
+                break;
+            }
+        }
+        std::make_heap(n._shards.begin(), n._shards.end(), shard_load_cmp());
```

Comment:
Adjusting a single shard should cost log n, not 3n.

Also, we have std::ranges::make_heap now.

Reply:
Fixed in V2